app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -81,24 +81,18 @@ def create
       object = @system || @activation_key || @distributor
 
       if params[:subscriptions]
-        params[:subscriptions].each do |sub|
-          subscription = Pool.find(sub[:id])
-          object.subscribe(subscription.cp_id, subscription[:quantity])
+        params[:subscriptions].each do |sub_params|
+          subscription = Pool.find(sub_params[:id])
+          object.subscribe(subscription.cp_id, sub_params[:quantity])
+          subscription.import_data
```

Comment:
If I read this correctly, now when we do an attach of a subscription to an object, we are doing an attach which calls out to Candlepin and then doing a data import from Candlepin to ensure out database is correct w.r.t. the Candlepin data?